util/movement-types/src/lib.rs

```diff
@@ -22,34 +22,44 @@ impl Id {
 	}
 }
 
+impl AsRef<[u8]> for Id {
+	fn as_ref(&self) -> &[u8] {
+		&self.0
+	}
+}
+
 impl fmt::Display for Id {
 	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
 		write!(f, "{:?}", &self.0)
 	}
 }
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct Transaction(pub Vec<u8>);
+pub struct Transaction {
+	pub data: Vec<u8>,
+	pub sequence_number: u64,
+}
 
 impl From<Vec<u8>> for Transaction {
 	fn from(data: Vec<u8>) -> Self {
-		Self(data)
+		Self { data, sequence_number: 0 }
 	}
 }
```

Comment:
This infallible conversion is a bit too risque to begin with, and now it's not clear that the sequence number 0 is the result of an `.into()`. I think it's better to remove this `From` impl and use the explicit constructor everywhere.

Reply:
Okay